### ISSUE TITLE
Link error of Optimal_transportation_reconstruction_2_demo when `Cimg` is found

### ISSUE THREAD

**Issue by lrineau** [CGAL/cgal#965]
On my machine, a build of all demos fails on one demo:
```
[ 36%] Linking CXX executable Optimal_transportation_reconstruction_2_demo
/usr/bin/ld: CMakeFiles/Optimal_transportation_reconstruction_2_demo.dir/window.cpp.o: undefined reference to symbol 'pthread_mutex_trylock@@GLIBC_2.2.5'
```

My machine has CImg.

Cc @CGAL/geometryfactory and the authors (@cjamin @palliez)

**Comment by cjamin** [comment]
If I understand correctly, you don't have the problem if you disable CImg?

**Comment by cjamin** [comment]
Actually, we already discussed this by e-mail on February 14.

The CImg doc says to link with pthread to build with g++: http://cimg.eu/reference/group__cimg__overview.html

Can you please test it?

**Comment by lrineau** [comment]
That's right. If, in the `CMakeCache.txt`, I set:
```cmake
CIMG_INCLUDE_DIR:PATH=
```
then the compilation has no error.

**Comment by lrineau** [comment]
If seems the correct way to use pthread on Linux is the `Threads` module of CMake

* http://stackoverflow.com/a/4774027/1728537 (for CMake<3.1)
* http://stackoverflow.com/a/29871891/1728537 (for CMake>=3.1), with the virtual target `Threads::Threads`

**Comment by lrineau** [comment]
If I have `CIMG_INCLUDE_DIR:PATH=/usr/include` in the cache, but add manual `-pthread` to the link command line, the compilation is fine too.

**Comment by sloriot** [comment]
@cjamin Any chance you can fix it before the end of the week or shall we postpone it to 4.9?

**Comment by cjamin** [comment]
It's in my TODO-list, but I won't have time to look into it this week. So let's post-pone it.

**Comment by sloriot** [comment]
@cjamin ping :)

**Comment by cjamin** [comment]
I'm still celebrating the header-only feature ! ;)
More seriously, I will look at those problems at the end of July. Hope that won't be too late.

**Comment by cjamin** [comment]
Fixed in #1322
